src/core/mac/sub_mac.cpp

```diff
@@ -349,26 +350,26 @@ void SubMac::StartCsmaBackoff(void)
     uint32_t backoffExponent = kMinBE + mTransmitRetries + mCsmaBackoffs;
 
 #if OPENTHREAD_CONFIG_MAC_CSL_TRANSMITTER_ENABLE
-    if (mTransmitFrame.mInfo.mTxInfo.mPeriod != 0)
+    if (mTransmitFrame.mInfo.mTxInfo.mTxDelay != 0)
     {
         SetState(kStateCslTransmit);
 
         if (ShouldHandleTransmitTargetTime())
         {
-            uint32_t phaseNow =
-                (otPlatRadioGetNow(&GetInstance()) / kUsPerTenSymbols) % mTransmitFrame.mInfo.mTxInfo.mPeriod;
-            uint32_t phaseDesired = mTransmitFrame.mInfo.mTxInfo.mPhase;
-
-            if (phaseNow < phaseDesired)
-            {
-                mTimer.Start((phaseDesired - phaseNow) * kUsPerTenSymbols);
-            }
-            else if (phaseNow > phaseDesired)
+            if (otPlatRadioGetNow(&GetInstance()) <
+                mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime + mTransmitFrame.mInfo.mTxInfo.mTxDelay)
             {
-                mTimer.Start((phaseDesired + mTransmitFrame.mInfo.mTxInfo.mPeriod - phaseNow) * kUsPerTenSymbols);
+#if OPENTHREAD_CONFIG_CSL_USE_PLATFORM_DELAY_TRANSMIT
+                BeginTransmit();
+#else
+                mTimer.StartAt(Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime),
+                               mTransmitFrame.mInfo.mTxInfo.mTxDelay);
+#endif
```

Comment:
When radio supports `OT_RADIO_CAPS_TRANSMIT_TIMING`, the direct transmission should falls to else part of `ShouldHandleTransmitTargetTime`.
```suggestion
                mTimer.StartAt(Time(mTransmitFrame.mInfo.mTxInfo.mTxDelayBaseTime),
                               mTransmitFrame.mInfo.mTxInfo.mTxDelay);
```